docs/src/developer-guide/working-with-rules.md

```diff
@@ -660,6 +660,8 @@ To learn more about JSON Schema, we recommend looking at some examples in [websi
 
 **Note:** Currently you need to use full JSON Schema object rather than array in case your schema has references ($ref), because in case of array format ESLint transforms this array into a single schema without updating references that makes them incorrect (they are ignored).
 
+**Note:** A common pitfall for JSON schema authors is to use `"oneOf"`, when `"anyOf"` could be used. Usually `"anyOf"` is preferable, because it’s more performant.
```

Comment:
it may be worth noting here that when describing all potential properties of an object, `oneOf` makes sense, but when describing a specific property, `anyOf` makes the most sense.